GVFS/GVFS.Virtualization/FileSystem/FileSystemVirtualizer.cs

```diff
@@ -202,6 +202,13 @@ protected void OnWorkingDirectoryFileOrFolderDeleteNotification(string relativeP
                         this.FileSystemCallbacks.OnFolderDeleted(relativePath);
                     }
                 }
+                else
+                {
+                    // During a git command if it deletes a folder we need to track that as a tombstone
+                    // So that we can delete them if the projection changes. This will be a no-op on platforms
+                    // that don't override OnPossibleTombstoneFolderCreated
+                    this.OnPossibleTombstoneFolderCreated(relativePath);

```

Comment:
One question I had - why is this a `possibleTombstoneFolder`? Here we have a notification that the directory was deleted, so we should "know" that the folder was deleted, and is a tombstone? When would we delete the directory here, but the directory would not be a tombstone?

Reply:
The folder could have been a full folder and we would still get a notification for the delete but there would not be a tombstone for the folder.